lib/otel/segments/utils.js

```diff
@@ -5,12 +5,22 @@
 
 'use strict'
 
+/**
+ * Accepts trace context payload if span has a parent. It will use the
+ * span context to extract the traceId, traceFlags and trace state.
+ *
+ * @param {object} params to function
+ * @param {Transaction} params.transaction active transaction
+ * @param {object} params.otelSpan active span
+ * @param {string} params.transport indicator of type of span(http, kafkajs, rabbitmq, etc)
+ */
 function propagateTraceContext({ transaction, otelSpan, transport }) {
   const spanContext = otelSpan.spanContext()
+  const parentSpanId = otelSpan.parentSpanId || otelSpan?.parentSpanContext?.spanId
```

Comment:
```suggestion
  const parentSpanId = otelSpan.parentSpanId || otelSpan.parentSpanContext?.spanId
```

or

```suggestion
  const parentSpanId = otelSpan?.parentSpanId || otelSpan?.parentSpanContext?.spanId
```

We either have the root object or we don't.